src/iosrv/srv.c

```diff
@@ -1802,3 +1803,67 @@ dss_dump_ABT_state()
 	}
 	ABT_mutex_unlock(xstream_data.xd_mutex);
 }
+
+#define GC_CREDS	128
+
+void
+dss_gc_run(int credits)
+{
+	int	total = 0;
+
+	while (1) {
+		int	creds = 128;
+		int	rc;
+
+		if (credits > 0 && (credits - total) < creds)
+			creds = credits - total;
+
+		total += creds;
+		rc = vos_gc_run(&creds);
+		if (rc) {
+			D_ERROR("GC run failed: %s\n", d_errstr(rc));
+			break;
+		}
+		total -= creds; /* subtract the remainded credits */
+		if (creds != 0) {
+			D_DEBUG(DB_MD, "GC consumed %d credits\n", total);
+			break;
+		}
+
+		if (credits > 0 && total >= credits)
+			break;
+
```

Comment:
Need to break the loop on server shutdown.

Reply:
indeed